### PR TITLE
[SPARK-13322] [ML] AFTSurvivalRegression supports feature standardization

### PR DESCRIPTION
**Pull request by yanboliang**
## What changes were proposed in this pull request?

AFTSurvivalRegression should support feature standardization, it will improve the convergence rate.
Test the convergence rate on the [Ovarian](https://stat.ethz.ch/R-manual/R-devel/library/survival/html/ovarian.html) data which is standard data comes with Survival library in R,
- without standardization(before this PR) -> 74 iterations.
- with standardization(after this PR) -> 38 iterations.

But after this fix, with or without `standardization` will converge to the same solution. It means that `standardization = false` will run the same code route as `standardization = true`. Because if the features are not standardized at all, it will result convergency issue when the features have very different scales. This behavior is the same as ML [`LinearRegression` and `LogisticRegression`](https://issues.apache.org/jira/browse/SPARK-8522). See more discussion about this topic at #11247. 
cc @mengxr 
## How was this patch tested?

unit test.

### CODE REVIEW

**Comment by mengxr** [comment]
cc @dbtsai

File: mllib/src/main/scala/org/apache/spark/ml/regression/AFTSurvivalRegression.scala [hunk 1]
```diff
@@ -230,7 +241,13 @@ class AFTSurvivalRegression @Since("1.6.0") (@Since("1.6.0") override val uid: S
 
     if (handlePersistence) instances.unpersist()
 
-    val coefficients = Vectors.dense(parameters.slice(2, parameters.length))
+    val rawCoefficients = parameters.slice(2, parameters.length)
+    var i = 0
+    while (i < numFeatures) {
+      rawCoefficients(i) *= { if (featuresStd(i) != 0.0) 1.0 / featuresStd(i) else 0.0 }
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/ml/regression/AFTSurvivalRegression.scala]: Could you test a dataset with a constant nonzero column and see whether we output a different solution from R? While I think it is correct to set the coefficients to 0.0 for a constant feature column. But if it is different from R's output, we should document it. See the discussion on SPARK-13029.

**Comment by yanboliang** [comment]
@mengxr I run test on a dataset with a constant nonzero column. 
* If ```fitIntercept==true```, Spark ```AFTSurvivalRegression``` and R ```survreg``` output the same result which setting the coefficients to 0.0 for the constant feature column.
* If ```fitIntercept==false```, Spark ```AFTSurvivalRegression``` output different solution from R ```survreg```. R output nonzero coefficients for the constant feature column.

If there are constant columns and ```fitIntercept``` is false. We should output a warning message and clarify in document. I think it should be handle with ```LogisticRegression``` together at [SPARK-13590](https://issues.apache.org/jira/browse/SPARK-13590).

**Comment by mengxr** [comment]
Sounds good. Assigned that ticket to you. I'm merging this PR into master. Thanks!

**Pull request closed without merging**